src/app/qgsappscreenshots.cpp

```diff
@@ -0,0 +1,155 @@
+/***************************************************************************
+  qgsappscreenshots.cpp
+  --------------------------------------
+  Date                 : September 2018
+  Copyright            : (C) 2018 by Denis Rouzaud
+  Email                : denis@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include <QMenu>
+#include <QWindow>
+#include <QScreen>
+#include <QImageWriter>
+
+#include "qgsappscreenshots.h"
+
+#include "qgsvectorlayerproperties.h"
+#include "qgsvectorlayer.h"
+#include "qgsproject.h"
+#include "qgsmessagelog.h"
+#include "qgisapp.h"
+
+QgsAppScreenShots::QgsAppScreenShots( const QString &saveDirectory )
+  : mSaveDirectory( saveDirectory )
+{
+  QString layerDef = QStringLiteral( "Point?crs=epsg:4326&field=pk:integer&field=my_text:string&field=my_integer:integer&field=my_double:double&key=pk" );
```

Comment:
We probably want some standardised, beautifully styled layers here?

Reply:
that would be _wunderbar_